### ISSUE TITLE
Some jobs being enqueued twice (Fixed in v1.1.1)

### ISSUE THREAD

**Issue by MSeneadza** [simplymadeapps/simple_scheduler#55]
Hi again guys,

First, I've been happily using the gem for over 2 years now.  It continues to be a great help to me.  So thanks again...

I upgraded from 1.0.0 to 1.1.0 on July 31st.  The next day one of my jobs (ProDigestsWorker) started running twice instead of the once daily it's been running for years.  I only noticed because it sends emails to my users and a couple of them started complaining that they were getting the nightly email twice, about 3 minutes apart.  

Today I watched the scheduled jobs as the time for the emails to be sent approached.  The list looked fine -- I only saw that job scheduled once within the upcoming hours.  So the job started -- it queries all the users who should get the email and then enqueues other jobs to send each individual email.  

So I watched that process work as it should but to my surprise, when I looked at the default queue, I see ProDigestsWorker enqueued again.  Do you have any guesses as to why this may be happening?

My first thought was that ProDigestsWorker was failing and being retried, but it's set to: `sidekiq_options queue: 'default', retry: false`  And it doesn't show up in my exception tracker. 

Now I'm thinking that maybe there's just a timing quirk with when Heroku Scheduler is running `rake simple_scheduler` -- ProDigestsWorker runs at 7:05 and Heroku Scheduler is *supposed* to run at :03, :13, etc.  ProDigestsWorker runs for about 2 minutes and I'm wondering if somehow the rake task is re-scheduling it while it's running.

Any thoughts would be appreciated.  I know this may be impossible for you to debug but I thought I'd ask in case one of the 1.1.0 changes may be causing this. 

Thanks again.

**Comment by brianpattison** [comment]
@MSeneadza Thanks for the report. We will have to look into this, but it is possible that the changes in 1.1.0 introduced a bug that could re-schedule the job. Did you make any changes to your simple_scheduler.yml config file when upgrading? And also, please send us the block of the simple_scheduler.yml config file for the ProDigestsWorker so I can try to reproduce this. Thanks again!

**Comment by MSeneadza** [comment]
Thanks @brianpattison   I had to add a few `at:` entries for some of my jobs but not for ProDigestsWorker.  The config for it wasn't changed.  Here's it is:

```
send_pro_digests:
  class: "ProDigestsWorker"
  every: "1.day"
  at: "19:05"
  expires_after: "2.hours"
  tz: "America/New_York"
```

**Comment by brianpattison** [comment]
@MSeneadza I am able to reproduce this issue. Please downgrade to v1.0.0 and I will ping you as soon as I release v1.1.1 with the fix. Thanks again!

**Comment by MSeneadza** [comment]
oh wow!  I'm glad I asked the question.  Thanks for the quick response.

**Comment by brianpattison** [comment]
@MSeneadza I reverted the code that backfills scheduled jobs so we can get a fix out for this immediately. This will solve your problem, but I'm going to leave this issue open for now until we make a decision on whether we want to bring some version of that code back in. It turns out it caused more issues than we could have imagined, so thank you so much for reporting this!!

Upgrading to v1.1.1 will fix your rescheduling issue.

**Comment by brianpattison** [comment]
The bug is fixed in v1.1.1. We're not looking to add this enhancement to backfill jobs at this time because of the issues it presented.
